hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/OMClientResponse.java

```diff
@@ -40,6 +41,21 @@ public OMClientResponse(OMResponse omResponse) {
     this.omResponse = omResponse;
   }
 
+  /**
+   * Check if omResponse status is OK. If yes, add to DB.
+   * For OmResponse with failure, this should do nothing. This method is not
+   * called in failure scenario in OM code.
+   * @param omMetadataManager
+   * @param batchOperation
+   * @throws IOException
+   */
+  public void checkAndUpdateDB(OMMetadataManager omMetadataManager,
```

Comment:
I like this approach of not checking getStatus check to OK, but I see we have some special case in S3MultipartUploadCommitPartResponse. I think in that case we need to override this method in S3MultipartUploadCommitPartResponse.
https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/s3/multipart/S3MultipartUploadCommitPartResponse.java#L70

Reply:
Yes. I will handle this when I get to Multipart idempotence. I noticed that in some requests, we missed checking status OK. So thought we should have it in the abstract class itself as it is common code also.

Reply:
But until we reach there, when this is merged current trunk is not correct right. As now in OzoneManagerDoubleBuffer, we call addToDBBatch only when the status is OK.

Reply:
Understood now. Thanks for catching this. I will update it.